x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/license/LicensingTests.java

```diff
@@ -210,49 +208,6 @@ public void testNodeJoinWithoutSecurityExplicitlyEnabled() throws Exception {
         }
     }
 
-    public void testWarningHeader() throws Exception {
```

Comment:
Why can't we keep this test? It validate that XPackLicenseState actually adds the headers.

Reply:
The message building part was moved to another test. However, I also have now added a test of this for XPackLicenseState, and it caught a bug (thanks!). Previously (even before this change) the warning was only emitted on a non-active license if the feature did not require an active license (ie a few security features).